tests/shared/core/training_data/test_structures.py

```diff
@@ -87,6 +87,38 @@ def test_as_story_string_or_statement():
     )
 
 
+def test_as_story_string_or_statement_with_slot_was_set():
+    import rasa.shared.utils.io
+
+    stories = """
+    stories:
+    - story: hello world
+      steps:
+      - or:
+        - slot_was_set:
+            - name: joe
+        - slot_was_set:
+            - name: bob
+      - action: some_action
+    """
+
+    reader = YAMLStoryReader(is_used_for_training=False)
```

Comment:
Do we need a test here when the `is_used_for_training` flag is `True` too?

Reply:
It's tested in another test (`test_or_statement_with_slot_was_set_is_used_for_training`) — in this case it makes no sense 'cause it will break the logic.

Reply:
What do you mean by it will break the logic? You mean the method `as_story_string()` can't be called for each of the 2 steps?

Reply:
Because when you specify that `is_used_for_training=True` then checkpoints are getting added and trackers are getting copied (see `add_stories` method in `StoryStepBuilder`, line 110). As a result, `reader.read_from_parsed_yaml(yaml_content)` will create multiple trackers and calling `.as_story_string` on them will lead to a result like that which is meaningless:

```
## hello world
    - slot{"name": "joe"}
> GENR_OR_b2ae7
```

However, this result is important when these trackers are used for training, and that case is already tested in `test_or_statement_with_slot_was_set_is_used_for_training`

Reply:
Great, thanks for the deep-dive!